shared/js/intl_helper.js

```diff
@@ -164,5 +200,12 @@
   }
 
   window.addEventListener('timeformatchange', global.IntlHelper, false);
-  window.addEventListener('languagechange', global.IntlHelper, false);
+
+  // We're using DOMLocalized here instead of languagechange because
+  // mozIntl formatters would restart before new language is loaded
+  // which means they'd be using old strings
```

Comment:
Sounds like a use case for `supportedlanguageschange`, although I'm happy with `DOMLocalized` too.